src/api/plans.ts

```diff
@@ -53,7 +53,19 @@ router.get("/plans/:planId", async (req, res) => {
   });
 });
 
-router.put("/plans/:planId", (req, res) => {
+router.put("/plans/:planId", async (req, res) => {
+  const user = await authUser(req);
+  const planId = req.params.planId;
+  const planBody = planReqBody.partial().parse(req.body);
+
+  await sql`
+    UPDATE plans SET
+    graduation_date = COALESCE(${planBody.graduationDate ?? null}, graduation_date),
+    concentration_id = COALESCE(${planBody.concentrationId ?? null}, concentration_id),
+    program_id = COALESCE(${planBody.programId ?? null}, program_id)
+    WHERE user_id = ${user.id} AND id = ${planId}
+  `;
```

Comment:
This whole thing is kinda weird, but basically I'm making it so that if a plan attribute isn't specified, it will keep the old value.

For example, if the payload `{concentrationId: 5}` is submitted, it will keep the program and graduation date the same, but change the concentration. Same thing applies with graduation date and program id.

Reply:
thats a great idea bestie!